File: utils/CGI/cgi.php
#!/usr/bin/php-cgi
<?php
// Read the input from the HTML form
$name = isset($_POST['name']) ? $_POST['name'] : '';

// Output the CGI response
echo "Content-type: text/html\r\n\r\n";
echo "<html><head><title>PHP CGI Example</title></head><body>";

if ($name) {
    echo "<h1>Hello, $name!</h1>";
} else {
    echo "<h1>Please enter your name in the form below.</h1>";
}

echo "<form method='post' action='index.php'>";
echo "Name: <input type='text' name='name'><br>";
echo "<input type='submit' value='Submit'>";
echo "</form>";

echo "</body></html>";
?>

File: test/cgi.py
import time

print("Content-type: text/html\n")
print("<html>")
print("<head>")
print("<title>Delayed CGI Example</title>")
print("</head>")
print("<body>")
print("<h1>Delayed Python CGI Example</h1>")
print("<p>This is a delayed response.</p>")

# Introduce a delay of 5 seconds
time.sleep(50)

print("<p>Delayed response is complete!</p>")
print("</body>")
print("</html>")